src/utils/HashMultiMap.fs

```diff
@@ -98,7 +98,7 @@ type internal HashMultiMap<'Key,'Value>(n: int, hasheq: IEqualityComparer<'Key>)
                     firstEntries.[y] <- h
                     rest.[y] <- t
                 | _ -> 
-                    // note: broken invariant
+                    // NOTE: broken invariant
```

Comment:
This comment seems a bit iffy.  If the invariant is broken ... swallowing it seems a poor choice.
Could you please review the code leading to this and make a proposal how to deal with it.